### PR TITLE
add workflows

### PR DESCRIPTION
**Pull request by yshrsmz**
tested on my repo

### CODE REVIEW

File: .github/workflows/pullrequest.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+name: Pull Request
```
Comment (yshrsmz) [inline .github/workflows/pullrequest.yml:1]: log for pull request workflow

https://github.com/yshrsmz/android-client-sdk/actions/runs/3089683437

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
+name: Build & Integration tests
```
Comment (yshrsmz) [inline .github/workflows/build.yml:1]: log form main branch on-push workflow

https://github.com/yshrsmz/android-client-sdk/actions/runs/3089932156

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        run: ./gradlew :bucketeer:testDebugUnitTest --stacktrace
+
+      - name: upload bucketeer build reports
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: bucketeer-build-reports
+          path: bucketeer/build/reports
+
+  androidTest:
```
Comment (yshrsmz) [inline .github/workflows/build.yml:43]: The workflow for the main branch will run both unit tests and integration tests

File: .github/workflows/pullrequest.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        run: ./gradlew lintKotlin --stacktrace
+
+      - name: Check Lint
+        run: ./gradlew lintDebug --stacktrace
+
+      - name: Build all build types
+        run: ./gradlew assemble --stacktrace
+
+      - name: Run Bucketeer unit tests
+        run: ./gradlew :bucketeer:testDebugUnitTest --stacktrace
```
Comment (yshrsmz) [inline .github/workflows/pullrequest.yml:39]: The workflow for pull-request will run lint, assemble, and unit tests. Integration tests will be executed on main branch.

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    needs: build
+    runs-on: macos-latest # enables hardware acceleration in the virtual machine
```
[the remark this answers is not in the document]
Comment (yshrsmz) [reply .github/workflows/build.yml:45]: note: macos is 10 times more costly than ubuntu.
https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#per-minute-rates

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+      - name: AVD cache
+        uses: actions/cache@v3
+        id: avd-cache
+        with:
+          path: |
+            ~/.android/avd/*
+            ~/.android/adb*
+          key: avd-29
+      - name: Run instrumentation tests
+        uses: reactivecircus/android-emulator-runner@v2
```
Comment (yshrsmz) [inline .github/workflows/build.yml:68]: I'm going to switch to Gradle Managed Device once we update AGP to 7.3.x
GMD should be much more reliable.

https://developer.android.com/studio/test/gradle-managed-devices


https://github.com/android/sunflower/pull/785
https://github.com/android/nowinandroid/pull/210

File: .github/workflows/pullrequest.yml (hunk 1 above)
Comment (cre8ivejp) [reply .github/workflows/pullrequest.yml:1]: It seems to be a bit faster than Bitrise, but the on-push workflow takes more than I imagined.

**Review by cre8ivejp — APPROVED** [review]
Finally, we can migrate all SDK CIs to the same platform :)
Thank you!

**Comment by yshrsmz** [comment]
Thanks!

**Pull request merged**